### ISSUE TITLE
Feature Request: Syntacic sugar for looping over a list of viewModels

### ISSUE THREAD

**Issue by RobertDiebels** [aurelia/templating#237]
Hi everyone!

I've got a feature request concerning ```<compose>```. Let's say I have a viewModel/view like so:

somemodel.js
```javascript
import {SomeOtherModel} from './someothermodel';

export class SomeModel{
    constructor() {
        this.items = [new SomeOtherModel('stuff', 'more stuff'), new SomeOtherModel('stuff', 'more stuff')];
    }
}
```
somemodel.html
```html
<template>
   <compose view="./someothermodel.html" repeat.for="item of items"></compose>
</template>
```
And some other viewModel/view like this:

someothermodel.js
```javascript
export class SomeOtherModel{
    constructor(param1, param2) {
        this.param1 = param1;
        this.param2 = param2;
    }
}
```
someothermodel.html
```html
<template>
    <div>
        <p>${item.param1}</p>
        <p>${item.param2}</p>
    </div>
</template>
```
What I would like is to request is some syntactic sugar so that I can shape the models/views into this: 

somemodel.js (remains the same)

somemodel.html
```html
<template>
   <someothermodel repeat.for="items"></someothermodel >
</template>
```
And some other viewModel/view like this:

someothermodel.js (remains the same)

someothermodel.html
```html
<template>
    <div>
        <p>${param1}</p>
        <p>${param2}</p>
    </div>
</template>
```
Or this:

somemodel.js (remains the same)

somemodel.html
```html
<template>
<template repeat.for="items">
   <someothermodel></someothermodel>
</template>
</template>
```
someothermodel.js (remains the same)
someothermodel.html (same as the version above)

Or this:

somemodel.js (remains the same)

somemodel.html
```html
<template>
<template repeat.for="item of items">
   <someothermodel view-model="item"></someothermodel>
</template>
</template>
```
someothermodel.js (remains the same)
someothermodel.html (same as the version above)

Basically it's a static version of ```<compose>``` and a form of syntactic sugar. If you'd want dynamic behaviour you would still use compose.

Would love to hear from you.

Greetings, Robert.

**Comment by EisenbergEffect** [comment]
Something very close to what you want is possible today. You can accomplish it by requiring the html directly in the view. If you change your view to this:

someothermodel.html
```html
<template bindable="viewModel">
    <div>
        <p>${viewModel.param1}</p>
        <p>${viewModel.param2}</p>
    </div>
</template>
```

Then you can do this:

somemodel.html
```htm
<template>
   <require from="someothermodel.html"></require>
   <template repeat.for="item of items">
      <someothermodel view-model="item"></someothermodel>
   </template>
</template>
```

Requiring an .html file into a view automatically converts it into a custom element with any bindable properties declared in the `bindable` attribute on the template.
